### PR TITLE
fixes orphaned ottca enrollments

### PR DESCRIPTION
**Pull request by andrewpmartinez**
- deletes ottca enrollment when CA is deleted
- adds API tests
- adds migration to remove currently orphaned ottca enrollments
- migration tested on a v15 database with orphaned ottca enrollments

### CODE REVIEW

File: tests/ca_test.go [hunk 1]
```diff
@@ -281,4 +282,69 @@ func Test_CA(t *testing.T) {
...
+				ctx.Req.NotEmpty(unenrolledOttCaIdentity.Id)
+				_ = ctx.AdminSession.requireQuery(fmt.Sprintf(`/identities?filter=id="%s"`, unenrolledOttCaIdentity.Id))
+			})
+
+			t.Run("identities with previous enrollments tied to deleted CAs should not error on detail", func(t *testing.T) {
+				ctx.testContextChanged(t)
+
+				ctx.Req.NotEmpty(unenrolledOttCaIdentity.Id)
+				_ = ctx.AdminSession.requireQuery("/identities/" + unenrolledOttCaIdentity.Id)
+			})
```
Comment (dovholuknf) [inline tests/ca_test.go:347]: should the test also verify that the sessions are removed? (maybe it does but i just don't see it)

File: controller/persistence/migration_v16.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			if err != nil && boltz.IsErrNotFoundErr(err) {
+				enrollmentsToDelete = append(enrollmentsToDelete, currentEnrollmentId)
+			}
+		}
+	}
+
+	//clear caIds that are invalid via CheckIntegrity
+	m.stores.Enrollment.CheckIntegrity(step.Ctx.Tx(), true, func(err error, fixed bool) {
+		if !fixed {
+			pfxlog.Logger().Errorf("unfixable error during orphaned ottca enrollment integrity check: %v", err)
```
Comment (dovholuknf) [inline controller/persistence/migration_v16.go:44]: is there any way to know the id of the unfixable cas? hopefully that's logged somewhere?

File: tests/ca_test.go (hunk 1 above)
Comment (ekoby) [reply tests/ca_test.go:347]: query would fail if not removed

**Review by dovholuknf — APPROVED** [review]

File: controller/persistence/migration_v16.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	"fmt"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/foundation/storage/ast"
+	"github.com/openziti/foundation/storage/boltz"
+)
+
+func (m *Migrations) removeOrphanedOttCaEnrollments(step *boltz.MigrationStep) {
+
+	var enrollmentsToDelete []string
+
```
Comment (plorenz) [inline controller/persistence/migration_v16.go:13]: You can use ast.BoolNodeTrue as a filter if you want everything

**Review by plorenz — APPROVED** [review]

**Review by ekoby — APPROVED** [review]

File: controller/persistence/migration_v16.go (hunk 3 above)
Comment (andrewpmartinez) [reply controller/persistence/migration_v16.go:13]: Nice. Will do.

File: controller/persistence/migration_v16.go (hunk 2 above)
Comment (andrewpmartinez) [reply controller/persistence/migration_v16.go:44]: The error that comes out from `CheckIntegrity` will state ids.

File: tests/ca_test.go (hunk 1 above)
Comment (andrewpmartinez) [reply tests/ca_test.go:347]: ekoby is correct!

**Pull request merged**